File: Rust/src/solution/most_frequent_even.rs
use crate::solution::Solution;
use std::collections::HashMap;

impl Solution {
    pub fn most_frequent_even(nums: Vec<i32>) -> i32 {
	let mut count_map: HashMap<i32, i32> = HashMap::new();
	for num in nums {
	    if num % 2 == 0 {
		*count_map.entry(num).or_insert(0) += 1;
	    }
	}
	let mut most_frequent = -1;
	let mut max_count = 0;
	for (num, count) in count_map {
	    if (count > max_count) || ((count == max_count) && (num < most_frequent)) {
		most_frequent = num;
		max_count = count;
	    }
	}
	most_frequent
    }
}

File: Dart/lib/mostFrequentEven.dart
class Solution {
  int mostFrequentEven(List<int> nums) {
    Map<int, int> count = {};
    for (int num in nums) {
      if (num % 2 == 0) {
        count[num] = (count[num] ?? 0) + 1;
      }
    }
    int maxCount = 0;
    int result = -1;
    for (var entry in count.entries) {
      if ((entry.value > maxCount) ||
          ((entry.value == maxCount) && (entry.key < result))) {
        maxCount = entry.value;
        result = entry.key;
      }
    }
    return result;
  }
}
